docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -73,14 +76,50 @@ The valid values are:
 Capping replicated table size
 -----------------------------
 
-Join distribution type will be chosen automatically when join reordering strategy
-is set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``.
-In such case it is possible to cap the maximum size of replicated table via
-``join-max-broadcast-table-size`` config property (e.g. ``join-max-broadcast-table-size=100MB``)
-or via ``join_max_broadcast_table_size`` session property (e.g. ``set session join_max_broadcast_table_size='100MB';``)
-This allows to improve cluster concurrency and to prevent bad plans when CBO misestimates size of joined tables.
+The join distribution type is automatically chosen when the join reordering
+strategy is set to ``AUTOMATIC`` or when the join distribution type is set to
+``AUTOMATIC``. In both cases, it is possible to cap the maximum size of the
+replicated table with the ``join-max-broadcast-table-size`` configuration
+property or with the ``join_max_broadcast_table_size`` session property. This
+allows you to improve cluster concurrency and prevent bad plans when the
+cost-based optimizer misestimates the size of the joined tables.
+
+By default, the replicated table size is capped to 100MB.
+
+Syntactic join order
```

Comment:
If I read this description correctly it describes the case with right most join tree, where no explicit parens are put to shape join order. For queries like this (I am using cross joins for brevity):
```
select * from nation n2 cross join nation n2 cross join nation n3 cross join nation n4 cross join nation n5;
```
which produce plan like:
```
 Output[]
 └─ CrossJoin[]
    ├─ CrossJoin[]
    │  ├─ CrossJoin[]
    │  │  ├─ CrossJoin[]
    │  │  │  ├─ TableScan[table = memory:6]
    │  │  │  └─ LocalExchange[partitioning = SINGLE]
    │  │  │     └─ RemoteExchange[type = REPLICATE]
    │  │  │        └─ TableScan[table = memory:6]
    │  │  └─ LocalExchange[partitioning = SINGLE]
    │  │     └─ RemoteExchange[type = REPLICATE]
    │  │        └─ TableScan[table = memory:6]
    │  └─ LocalExchange[partitioning = SINGLE]
    │     └─ RemoteExchange[type = REPLICATE]
    │        └─ TableScan[table = memory:6]
    └─ LocalExchange[partitioning = SINGLE]
       └─ RemoteExchange[type = REPLICATE]
          └─ TableScan[table = memory:6]
```

while query:
```
select * from (nation n2 cross join nation n2) cross join (nation n3 cross join nation n4 cross join nation n5)
```
produces
```
 Output[]
 └─ CrossJoin[]
    ├─ CrossJoin[]
    │  ├─ TableScan[table = memory:6]
    │  └─ LocalExchange[partitioning = SINGLE]
    │     └─ RemoteExchange[type = REPLICATE]
    │        └─ TableScan[table = memory:6]
    └─ LocalExchange[partitioning = SINGLE]
       └─ RemoteExchange[type = REPLICATE]
          └─ CrossJoin[]
             ├─ CrossJoin[]
             │  ├─ TableScan[table = memory:6]
             │  └─ LocalExchange[partitioning = SINGLE]
             │     └─ RemoteExchange[type = REPLICATE]
             │        └─ TableScan[table = memory:6]
             └─ LocalExchange[partitioning = SINGLE]
                └─ RemoteExchange[type = REPLICATE]
                   └─ TableScan[table = memory:6]
```

In the latter case you are not limited to executing a single join at a time.